src/inmanta/parser/plyInmantaLex.py

```diff
@@ -56,13 +55,56 @@
     "else",
     "elif",
 ]
-literals = [":", "[", "]", "(", ")", "=", ",", ".", "{", "}", "?", "*"]
+literals = [":", "[", "]", "(", ")", "=", ",", ".", "{", "}", "?", "*", "%"]
 reserved = {k: k.upper() for k in keyworldlist}
 
 # List of token names.   This is always required
-tokens = ["INT", "FLOAT", "ID", "CID", "SEP", "STRING", "MLS", "CMP_OP", "REGEX", "REL", "PEQ", "RSTRING", "FSTRING"] + sorted(
-    list(reserved.values())
-)
+tokens = [
+    "INT",
+    "FLOAT",
+    "ID",
+    "CID",
+    "SEP",
+    "STRING",
+    "MLS",
+    "CMP_OP",
+    "REGEX",
+    "MATCHING",
+    "REL",
+    "PEQ",
+    "RSTRING",
+    "FSTRING",
+    "PLUS_OP",
+    "MINUS_OP",
+    "DIVISION_OP",
+    "DOUBLE_STAR",
+] + sorted(list(reserved.values()))
+
+
+def t_REGEX(t: lex.LexToken) -> lex.LexToken:  # noqa: N802
+    r"matching[\s]+/([^/\\\n]|\\.)+/"
+    index_first_slash_char = t.value.index("/")
+    regex_with_slashes_as_str = t.value[index_first_slash_char:]
+    regex_as_str = regex_with_slashes_as_str[1:-1]
+    value = Reference("self")  # anonymous value
+    try:
+        expr = Regex(value, regex_as_str)
+        t.value = expr
+        return t
+    except RegexError as error:
+        t.value = regex_with_slashes_as_str
+        end = t.lexer.lexpos - t.lexer.linestart + 1
+        (s, e) = t.lexer.lexmatch.span()
+        start = end - (e - s) + index_first_slash_char
```

Comment:
I'm not sure this is accurate if there's a newline betweeen `matching` and the start of the regex. Same for the lineno below.

Reply:
The lexer doesn't allow a newline between the matching and the start of the regex, right?

Reply:
I made sure the line numbers are reported correctly when there is a newline between the matching keyword and the regex.